pkg/compute/deploy.go

```diff
@@ -320,95 +326,57 @@ func pkgPath(path string, name string, source manifest.Source) (string, error) {
 
 // validateService checks if the service version has a domain and backend
 // defined.
-func validateService(serviceID string, client api.Interface, version common.OptionalInt) (*fastly.Version, bool, invalidResource, error) {
-	v, err := serviceVersion(serviceID, client, version)
+//
+// The use of `resourceNone` as a `invalidResource` enum type is to represent a
+// situation where an error occurred and so we were unable to identify whether
+// a domain or backend (or both) were missing from the given service.
+func validateService(serviceID string, client api.Interface, version *fastly.Version) (bool, invalidResource, error) {
+	err := checkServiceID(serviceID, client, version)
 	if err != nil {
-		return nil, false, resourceNone, err
+		return false, resourceNone, err
 	}
 
 	domains, err := client.ListDomains(&fastly.ListDomainsInput{
 		ServiceID:      serviceID,
-		ServiceVersion: v.Number,
+		ServiceVersion: version.Number,
 	})
 	if err != nil {
-		return v, false, resourceNone, fmt.Errorf("error fetching service domains: %w", err)
+		return false, resourceNone, fmt.Errorf("error fetching service domains: %w", err)
 	}
 
 	backends, err := client.ListBackends(&fastly.ListBackendsInput{
 		ServiceID:      serviceID,
-		ServiceVersion: v.Number,
+		ServiceVersion: version.Number,
 	})
 	if err != nil {
-		return v, false, resourceNone, fmt.Errorf("error fetching service backends: %w", err)
+		return false, resourceNone, fmt.Errorf("error fetching service backends: %w", err)
 	}
 
 	ld := len(domains)
 	lb := len(backends)
 
 	if ld == 0 && lb == 0 {
-		return v, false, resourceBoth, nil
+		return false, resourceBoth, nil
 	}
 	if ld == 0 {
-		return v, false, resourceDomain, nil
+		return false, resourceDomain, nil
 	}
 	if lb == 0 {
-		return v, false, resourceBackend, nil
+		return false, resourceBackend, nil
 	}
 
-	return v, true, resourceNone, nil
+	return true, resourceNone, nil
 }
 
-// serviceVersion returns the version for the given service.
-func serviceVersion(serviceID string, client api.Interface, versionFlag common.OptionalInt) (*fastly.Version, error) {
+// checkServiceID validates the given Service ID maps to a real service.
+func checkServiceID(serviceID string, client api.Interface, version *fastly.Version) error {
```

Comment:
The `compute deploy` command was originally implementing similar logic to what is being proposed in this PR. Specifically it would attempt to acquire an 'ideal' service version to deploy a package into. This logic has been superseded by the new overloaded `--version` and `--autoclone` flag parsers.